From a9a881a2f768e331b1642c6a5e77758ad965bdea Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:07:32 -0400
Subject: [tools] perf/session: Fix possible null pointer dereference in session.c

Message-id: <1410181752-25631-208-git-send-email-jolsa@redhat.com>
Patchwork-id: 91971
O-Subject: [PATCH RHEL7.1 BZ1134356 207/307] perf session: Fix possible null pointer dereference in session.c
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit c5765ece8a050836c6255e1276fc8e0e867078da
Author: Masanari Iida <standby24x7@gmail.com>
Date: Thu May 15 02:13:38 2014 +0900

description
===========
cppcheck detected following warning:
[tools/perf/util/session.c:1628] -> [tools/perf/util/session.c:1632]:
 (warning) Possible null pointer dereference: session - otherwise it
 is redundant to check it against null.

In order to avoide null pointer, check the pointer before use.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/session.c b/tools/perf/util/session.c
index 55960f2..64a186e 100644
--- a/tools/perf/util/session.c
+++ b/tools/perf/util/session.c
@@ -1625,13 +1625,14 @@ out_delete_map:
 void perf_session__fprintf_info(struct perf_session *session, FILE *fp,
     bool full)
 {
- int fd = perf_data_file__fd(session->file);
  struct stat st;
- int ret;
+ int fd, ret;
 
  if (session == NULL || fp == NULL)
   return;
 
+ fd = perf_data_file__fd(session->file);
+
  ret = fstat(fd, &st);
  if (ret == -1)
   return;
-- 
1.7.1